festim/hydrogen_transport_problem.py

```diff
@@ -480,10 +481,39 @@ def create_dirichletbc_form(self, bc):
 
         return form
 
+    def define_sources(self):
+        """Create a fenics object value for each source of the model"""
+        for source in self.sources:
+            for idx, spe in enumerate(source.species):
```

Comment:
The approach taken here is to always have the source.species (same for volume) as a list. However this is modifying the user input quite a lot.

Instead, I think it would be better to do this modification here.

Something like:

```python
if not isinstance(source.species, list):
    source_species = [source.species]
else:
    source_species = source.species

for idx, spe in enumerate(species):
    ....
```

Does htat make sense?

Same is applicable to volume